bin/ch-test

```diff
@@ -212,11 +213,14 @@ test_examples () {
     echo 'executing example phase tests ...'
     for i in $examples; do
         if (test_make_check_image "$i"); then
+            bat_file=$(dirname "$i")/test.bats
             tag=$("${CHTEST_DIR}/chtest-make-auto" tag "$i")
-            export CH_TEST_TAG=$tag
-            bats "$(dirname "$i")/test.bats"
+            cp "$bat_file" "${tmp_}/${tag}_example.bats"
+            sed -i "s@\${ch_tag}@$tag@g" "${tmp_}/${tag}_example.bats"
+            sed -i "s@\$ch_tag@$tag@g" "${tmp_}/${tag}_example.bats"
```

Comment:
Avoid overloading shell variable syntax, because the mode of substitution will be unexpected to people reading the code later. Common syntaxes are `%foo%` and `@bar@`; I believe we may use this already in the template files?

This will also save a `sed` line.